topo/node/cisco/cisco.go

```diff
@@ -198,18 +198,92 @@ func fmtInt400(eid int) string {
 	return fmt.Sprintf("FourHundredGigE0/0/0/%d", eid)
 }
 
-func getCiscoInterfaceId(pb *tpb.Node, eth string) (string, error) {
-	ethWithIdRegx := regexp.MustCompile(`e(t(h(e(r(n(e(t)*)*)*)*)*)*)\d+`) // check for e|et|eth|....
+func setE8000Env(pb *tpb.Node) error {
+	interfaceList := "MgmtEther0/RP0/CPU0/0"
+	interfaceMap := "MgmtEther0/RP0/CPU0/0:eth0"
+	eths := make([]string, 0)
+	for k := range pb.Interfaces {
+		eths = append(eths, k)
+	}
+	sort.Strings(eths)
+	for _, eth := range eths {
+		ciscoInterfaceID, err := getCiscoInterfaceID(pb, eth)
+		if err == nil {
+			interfaceList = fmt.Sprintf("%s,%s", interfaceList, ciscoInterfaceID)
+			interfaceMap = fmt.Sprintf("%s,%s:%s", interfaceMap, ciscoInterfaceID, eth)
+		} else {
+			return err
+		}
+	}
+	if pb.Config.Env == nil {
```

Comment:
code duplication can be avoided here:

if pb.Config.Env == nil {
  pb.Config.Env = map[string]string{}
}
if pb.Config.Env["XR_INTERFACES"] == "" {
  pb.Config.Env["XR_INTERFACES"] = interfaceMap
}
if pb.Config.Env["XR_CHECKSUM_OFFLOAD_COUNTERACT"] == "" {
  pb.Config.Env["XR_CHECKSUM_OFFLOAD_COUNTERACT"] = interfaceList
}
if pb.Config.Env["XR_EVERY_BOOT_CONFIG"] == "" {
  pb.Config.Env["XR_EVERY_BOOT_CONFIG"] = filepath.Join(pb.Config.ConfigPath, pb.Config.ConfigFile)
}

Reply:
done

Reply:
sorry, I missed this one. Now It is fixed.